src/sc/sidechain.cpp

```diff
@@ -241,12 +251,36 @@ bool Sidechain::checkTxSemanticValidity(const CTransaction& tx, CValidationState
                     CValidationState::Code::INVALID, "sidechain-sc-creation-invalid-constant");
         }
 
-        if (sc.wCeasedVk.is_initialized() && !sc.wCeasedVk.get().IsValid())
+        if (sc.wCeasedVk.is_initialized() )
         {
-            return state.DoS(100,
+            if (!sc.wCeasedVk.get().IsValid())
```

Comment:
Does VK "isValid" method logic depend on `ProvingSystemType`?
For example, Vk was created for Darlin system, so we expect the key to be valid for Darlin and not valid for CoboundaryMarlin. Or is it valid for both?
Does it make any sense? @DanieleDiBenedetto